src/main/java/com/thoughtworks/gauge/Table.java

```diff
@@ -95,8 +102,90 @@ public void addRow(List<String> row) {
     
     @Override
     public String toString(){
-        return "Table{" +
-                        tableRows.toString() +
-                        '}';
+        int maxStringLength=getMaxStringLength();
+        List<String> formattedHeaderAndRows = new ArrayList<String>();
+        addHeader(maxStringLength, formattedHeaderAndRows);
+        addDashes(maxStringLength, formattedHeaderAndRows);
+        addValues(maxStringLength, formattedHeaderAndRows);
+        return Joiner.on(System.getProperty("line.separator")).join(formattedHeaderAndRows);
+    }
+
+    private void addDashes(
+        int maxStringLength,
+        List<String> formattedHeaderAndRows) {
+
+        String dashesString = Joiner.on("").join(Collections.nCopies(maxStringLength, "-"));
+        List<String> dashes = Collections.nCopies(headers.size(), dashesString);
+        String formattedDashes = formattedRow(dashes, maxStringLength);
+        formattedHeaderAndRows.add(formattedDashes);
+    }
+
+    private void addHeader(
+        int maxStringLength,
+        List<String> formattedHeaderAndRows) {
+
+        String formattedHeaders = formattedRow(headers, maxStringLength);
+        formattedHeaderAndRows.add(formattedHeaders);
+    }
+
+    private void addValues(
+        int maxStringLength,
+        List<String> formattedHeaderAndRows) {
+
+        for(TableRow tableRow:tableRows){
+            formattedHeaderAndRows.add(formattedRow(tableRow.getCellValues(), maxStringLength));
+        }
+    }
+    
+    private String formattedRow(List<String> strings,int maxStringLength){
+        List<String> formattedStrings = Lists.transform(strings, format(maxStringLength));
+        return "|"+Joiner.on("|").join(formattedStrings)+"|";
+    }
+
+    private Function<String, String> format(
+        final int maxStringLength) {
+
+        
+        return new Function<String, String>() {
+
+            @Override
+            public String apply(
+                String input) {
+
+                
+                return " "+Strings.padEnd(input, maxStringLength, " ".charAt(0))+" ";
```

Comment:
Can we remove the spaces at both ends of each cell? So that, we have consistency in the way Gauge formats table and the way we print table in Java plugin.